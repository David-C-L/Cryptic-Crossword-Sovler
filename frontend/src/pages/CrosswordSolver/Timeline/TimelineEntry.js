function TimelineEntry(props) {

  return (
    <div>
      
      <hr style={{borderTop: '1px solid #8c8b8b', margin: 0, padding: 0}}/>
      {props.entry &&
      <p><strong>{props.entry.location}</strong> is <strong>{props.entry.answer}</strong> because {props.entry.explanation} with confidence <strong>{props.entry.confidence}%</strong></p>
      }
    </div>
  )
} 

export default TimelineEntry
