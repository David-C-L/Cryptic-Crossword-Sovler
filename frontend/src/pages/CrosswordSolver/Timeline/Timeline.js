import TimelineEntry from "./TimelineEntry";

function Timeline(props) {

  return (
    <div style={{ 'overflow-y': 'scroll', 'maxHeight': '80vh' }}>
      <h3 className="text-center">Timeline</h3>
      {props.timeline && props.timeline.map(entry =>
        <TimelineEntry entry={entry} />
      )
      }
    </div>

  )
}

export default Timeline