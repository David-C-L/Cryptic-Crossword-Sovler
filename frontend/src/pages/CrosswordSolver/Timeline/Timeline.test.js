import React from "react";
import { unmountComponentAtNode } from "react-dom";
import renderer from 'react-test-renderer';
import Timeline from "./Timeline";
import { BrowserRouter as Router } from "react-router-dom";


let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('renders correctly', () => {
  const tree = renderer
    .create(<Router><Timeline /></Router>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});