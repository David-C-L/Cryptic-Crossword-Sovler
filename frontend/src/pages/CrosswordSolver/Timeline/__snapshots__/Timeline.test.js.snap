// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "maxHeight": "80vh",
      "overflow-y": "scroll",
    }
  }
>
  <h3
    className="text-center"
  >
    Timeline
  </h3>
</div>
`;
